pkg/v3/ocr3.go

```diff
@@ -0,0 +1,64 @@
+package v3
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+type InstructionStore interface{}
+
+// SamplingStore at this point can be an empty interface
+type SamplingStore interface{}
+
+// ResultStore at this point can be an empty interface
+type ResultStore interface{}
+
+type ocr3Plugin struct {
+	PrebuildHooks     []func(AutomationOutcome) error
+	BuildHooks        []func(*AutomationObservation, InstructionStore, SamplingStore, ResultStore) error
+	InstructionSource InstructionStore
+	MetadataSource    SamplingStore
+	ResultSource      ResultStore
+}
+
+func (plugin *ocr3Plugin) Observation(ctx context.Context, outcome ocr3types.OutcomeContext, query types.Query) (types.Observation, error) {
+	// Decode the outcome to AutomationOutcome
+	automationOutcome, err := DecodeAutomationOutcome(outcome.PreviousOutcome)
+	if err != nil {
+		return nil, err
+	}
+
+	// Execute pre-build hooks
+	for _, hook := range plugin.PrebuildHooks {
+		err := hook(automationOutcome)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Create new AutomationObservation
+	observation := AutomationObservation{
+		Instructions: automationOutcome.Instructions,
+		Metadata:     automationOutcome.Metadata,
+		Performable:  automationOutcome.Performable,
```

Comment:
These values should not be set using the outcome. You can just create a new observation and leave all fields blank.